services/api/src/api/routes/endpoint.py

```diff
@@ -27,8 +29,20 @@
 )
 
 
-def create_processing_step_endpoint(
-    processing_step: ProcessingStep,
+class EndpointsDefinition:
+    """Definition of supported endpoints and its relation with processing steps."""
+
+    definition: Mapping[str, List[ProcessingStep]]
+
+    def __init__(self, graph: ProcessingGraph, endpoint_config: EndpointConfig):
+        self.definition = {
+            endpoint: [graph.get_step_by_job_type(step) for step in processing_steps]
```

Comment:
I know it's the same, but I think we refer to the step name here, so

```suggestion
            endpoint: [graph.get_step(step) for step in processing_steps]
```